lib/school/decorator.rb

```diff
@@ -0,0 +1,26 @@
+require_relative 'nameable'
+
+class Decorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
+  end
+end
```

Comment:
- [OPTIONAL] Please it is a best practice to place each class in a separate file, if you are following the ruby way, placing your classes in separate files scales better.  If you place multiple classes in the same file and they start to grow, then you're going to need to separate them later. Kindly place your classes in separate files